test/Microsoft.Azure.WebJobs.Host.UnitTests/FunctionIndexerFactory.cs

```diff
@@ -24,44 +24,38 @@ internal static class FunctionIndexerFactory
     {
         public static FunctionIndexer Create(CloudStorageAccount account = null, INameResolver nameResolver = null, IExtensionRegistry extensionRegistry = null, TraceWriter traceWriter = null)
         {
-            Mock<IServiceProvider> services = new Mock<IServiceProvider>(MockBehavior.Strict);
-            StorageClientFactory clientFactory = new StorageClientFactory();
-            services.Setup(p => p.GetService(typeof(StorageClientFactory))).Returns(clientFactory);
-            IStorageAccount storageAccount = account != null ? new StorageAccount(account, services.Object) : null;
-            IStorageAccountProvider storageAccountProvider = new SimpleStorageAccountProvider(services.Object)
-            {
-                StorageAccount = account
-            };
-            IExtensionTypeLocator extensionTypeLocator = new NullExtensionTypeLocator();
-            ContextAccessor<IMessageEnqueuedWatcher> messageEnqueuedWatcherAccessor =
-                new ContextAccessor<IMessageEnqueuedWatcher>();
-            ContextAccessor<IBlobWrittenWatcher> blobWrittenWatcherAccessor =
-                new ContextAccessor<IBlobWrittenWatcher>();
-            ISharedContextProvider sharedContextProvider = new SharedContextProvider();
-            TraceWriter logger = traceWriter ?? new TestTraceWriter(TraceLevel.Verbose);
+            IStorageAccountProvider storageAccountProvider = GetStorageAccountProvider(account);
+
+            var config = TestHelpers.NewConfig(storageAccountProvider, nameResolver, extensionRegistry);
```

Comment:
>TestHelpers.NewConfig [](start = 25, length = 21)

This function is another example of code that's redundant with JobHostConfiguration setup.  The key thing is that it now calls the TestHelpers.NewConfig() rather than duplicate all the logic and glue on its own.